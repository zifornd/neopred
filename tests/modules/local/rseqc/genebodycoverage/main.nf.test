nextflow_process {

    name "Test Process RSEQC_GENEBODYCOVERAGE"
    script "modules/local/rseqc/genebodycoverage/main.nf"
    process "RSEQC_GENEBODYCOVERAGE"

    test("rseqc genebodycoverage with bam files") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            process {
                """
                // define inputs of the process here. Example:
                input[0] = Channel.of([ [ id:'test' ], file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true), file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam.bai', checkIfExists: true)])
               input[1] = file('${projectDir}/assets/housekeeping_refseqGenes.bed')
                """
            }
        }

        then {
            assert process.success
	    assert process.out.rscript
            assert process.out.gbc_txt
            assert process.out.pdf_curves
            assert process.out.versions
            assert snapshot(process.out)//.match()
        }

    }

}
