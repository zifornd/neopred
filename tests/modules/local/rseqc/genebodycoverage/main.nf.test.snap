{
    "rseqc genebodycoverage with bam files": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.geneBodyCoverage.r:md5,c848e1a588b26ab7a75965b6accad448"
                    ]
                ],
                "1": [
                    "test.geneBodyCoverage.txt:md5,11ecc8f0c9644450685fe48ead48ffe4"
                ],
                "2": [
                    "test.geneBodyCoverage.curves.pdf:md5,e809c6cbfee3e0875dede920b9c62bce"
                ],
                "3": [
                    "versions.yml:md5,67be82cbc17b1b31081fee521a2a8ff1"
                ],
                "gbc_txt": [
                    "test.geneBodyCoverage.txt:md5,11ecc8f0c9644450685fe48ead48ffe4"
                ],
                "pdf_curves": [
                    "test.geneBodyCoverage.curves.pdf:md5,e809c6cbfee3e0875dede920b9c62bce"
                ],
                "rscript": [
                    [
                        {
                            "id": "test"
                        },
                        "test.geneBodyCoverage.r:md5,c848e1a588b26ab7a75965b6accad448"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,67be82cbc17b1b31081fee521a2a8ff1"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.1"
        },
        "timestamp": "2024-12-11T06:01:21.959497"
    }
}